Bottom: 1553e4515d43f27e90a387ea11955bc95a2ec811
Top:    6dd26e08b1a049e802beb0cfdf1c515c6e77e987
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-28 00:42:51 +0000

Refresh of try-prepending-the-stolen-when

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index d7a8f4a..3943473 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1661,9 +1661,10 @@ steal_object_geometry (object3d *dst, object3d *src)
 {
   g_assert (dst != src);
 
-  dst->faces    = g_list_concat (dst->faces,    src->faces);
-  dst->edges    = g_list_concat (dst->edges,    src->edges);
-  dst->vertices = g_list_concat (dst->vertices, src->vertices);
+  /* Prepend the src chunks, as src is likely to be a lot smaller then dst once we get going */
+  dst->faces    = g_list_concat (src->faces,    dst->faces);
+  dst->edges    = g_list_concat (src->edges,    dst->edges);
+  dst->vertices = g_list_concat (src->vertices, dst->vertices);
   src->faces = NULL;
   src->edges = NULL;
   src->vertices = NULL;
